library/classes/Document.class.php

```diff
@@ -1056,6 +1086,29 @@ function createDocument(
                     return xl('Failed to create') .  $filepath . $this->get_thumb_name($filenameUuid);
                 }
             }
+            return '';
+            }); //end of fiber
+            /**
+             * The fiber is started and is immediately suspended. Values are passed through the Fiber::suspend() method.
+             */
+            $data = $fiber->start();
+            $offSiteUpload = new PatientDocumentUpload($data);
+            $offSiteUpload->setRemoteFileName($filename);
+            /**
+             * There must be a return to terminate for the fiber if there is a listener that uploaded the file.
+             */
+            $uploadListener = $this->eventDispatcher->dispatch($offSiteUpload, PatientDocumentUpload::REMOTE_STORAGE_LOCATION);
+            $uploadReturn = json_decode(json_encode($uploadListener), true);  // convert symfony object to array
+            /**
+             * If the listener is not null then the file was uploaded to another location.
+             * Else resume the local file storage
+             */
+
+            if ($uploadReturn) {
+                return '';
```

Comment:
If something else uploads the file data, why are you skipping over saving the Document properties in lines 1124-1139?  Shouldn't the caller module return properties that will be saved in this routine for the Document? Otherwise, anyone using this class will end up with an invalid Document object.

Reply:
This is My first draft and I do welcome the comments and feedback.  Fiber is new and I am trying to use it in real world asynchronous application. I need to spend more time in the document class